pkg/apis/core/validation/validation.go

```diff
@@ -4362,17 +4362,25 @@ func validatePodResources(spec *core.PodSpec, podClaimNames sets.Set[string], fl
 	if spec.Resources == nil {
 		return nil
 	}
+	resourcesFldPath := fldPath.Child("resources")
+	if spec.OS != nil && spec.OS.Name == core.Windows {
+		// Do not include more detailed errors on the resources field value
+		// if the resources field cannot be set on the target OS.
+		return field.ErrorList{
+			field.Forbidden(resourcesFldPath, "cannot be set for a windows pod"),
```

Comment:
Nit: usually we want "may not" language.

Reply:
Thanks! Updated.